_assets/js/search.js

```diff
@@ -21,19 +21,39 @@
       timer = setTimeout(next, timeout > 0 ? timeout : 300);
       return timer;
     };
-  }
+  };
 
   //////////////////////////////////
   // Translation
 
-  const translate_fromSearchGov = (response) => {
-    return response.web.results.map(result => {
-      return {
-        url: result.url,
-        title: result.title,
-        description: result.snippet
-      };
-    });
+  const translate_fromSearchGov = (query) => (response) => {
+    return {
+      routeTo: (() => {
+        if (!response.route_to) {
+          return null;
+        }
+
+        // If this is a routed query pointing to production search results, extract
+        // search parameters out.
+        // So we can test lower environments against the production search.gov,
+        // assume production hostname.
+        const url = new URL(response.route_to);
+        if (url.host == 'faq.coronavirus.gov' && url.pathname == '/search/') {
+          // Return the URL minus the hostname, with the orginal query included
+          return `${BASE_URL}${url.pathname}${url.search}&source=${encodeURIComponent(query)}`;
+        }
+
+        // This is a routed query to something other than search results.
+        return response.route_to;
+      })(),
+      results: response.web && response.web.results ? response.web.results.map(result => {
```

Comment:
Could be written...
```javascript
response.web.results.map(({url, title, description}) => ({url, title, description}));
```

Reply:
True.. although `snippet` is the source for `description` here, so I think I'll keep it for explicitness.

Reply:
gotcha!